HotCommands/Commands/MoveMemberUp.cs

```diff
@@ -0,0 +1,98 @@
+﻿using System;
+using Microsoft.VisualStudio.Shell;
+using Microsoft.VisualStudio;
+using Microsoft.VisualStudio.Text.Editor;
+using System.Linq;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+
+namespace HotCommands
+{
+    /// <summary>
+    /// Command handler for ToggleComment
+    /// </summary>
+    internal sealed class MoveMemberUp
+    {
+        /// <summary>
+        /// VS Package that provides this command, not null.
+        /// </summary>
+        private readonly Package package;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="MoveMethodUp"/> class.
+        /// </summary>
+        /// <param name="package">Owner package, not null.</param>
+        private MoveMemberUp(Package package)
+        {
+            if (package == null)
+            {
+                throw new ArgumentNullException("package");
+            }
+
+            this.package = package;
+        }
+
+        /// <summary>
+        /// Gets the instance of the command.
+        /// </summary>
+        public static MoveMemberUp Instance
+        {
+            get;
+            private set;
+        }
+
+        /// <summary>
+        /// Gets the service provider from the owner package.
+        /// </summary>
+        private IServiceProvider ServiceProvider
+        {
+            get
+            {
+                return this.package;
+            }
+        }
+
+        /// <summary>
+        /// Initializes the singleton instance of the command.
+        /// </summary>
+        /// <param name="package">Owner package, not null.</param>
+        public static void Initialize(Package package)
+        {
+            Instance = new MoveMemberUp(package);
+        }
+
+        public int HandleCommand(IWpfTextView textView)
+        {
+            var memberslist = new[] { "MethodDeclaration", "PropertyDeclaration", "FieldDeclaration" };
+            MemberDeclarationSyntax prevMember = null;
+            MemberDeclarationSyntax member = null;
+            var cursorPosition = textView.Caret.ContainingTextViewLine.Extent.Span.Start;
+            var members = CSharpSyntaxTree.ParseText(textView.TextSnapshot.GetText()).GetRoot().DescendantNodes().OfType<MemberDeclarationSyntax>();
```

Comment:
A couple of other important points here:

* Don't assume that the surface buffer of the view is C#. If this is a razor file, this will sure to fail. C# in projection buffers should work.
* Parsing the code yourself means that you have to take the states of various `#if..#endif` preprocessor into account. This will almost certainly be incorrect in shared projects where the current project context must be taken into account.